controllers/templatesync/template_sync.go

```diff
@@ -296,13 +346,27 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
 
 		var rsrc schema.GroupVersionResource
 
+		// Detect ConstraintTemplate
+		isGkConstraintTemplate := gvk.Group == utils.GvkConstraintTemplate.Group &&
+			gvk.Kind == utils.GvkConstraintTemplate.Kind
+		isGatekeeper := isGkConstraintTemplate || gvk.Group == utils.GConstraint
+
 		mapping, err := rMapper.RESTMapping(gvk.GroupKind(), gvk.Version)
 
 		if mapping != nil {
 			rsrc = mapping.Resource
 		} else {
 			resultError = err
-			errMsg := fmt.Sprintf("Mapping not found, please check if you have CRD deployed: %s", err)
+			errMsg := "Mapping not found, "
+			if isGatekeeper {
+				errMsg += "check if Gatekeeper is installed"
+				if gvk.Group == utils.GConstraint {
+					errMsg += " and the required ConstraintTemplate has been deployed"
```

Comment:
If the Policy has the ConstraintTemplate and the constraint, there is going to be a delay before Gatekeeper creates the CRD based on the ConstraintTemplate. My preference is to detect if a ConstraintTemplate was created previously in the loop and adjust the message.

Reply:
I saw in your code update that the message is just "check if the required ConstraintTemplate has been deployed". Is that sufficient here or was there another message you were thinking about having?

I ask because I'm not sure there's a huge bump in value to adjusting the message based on whether the ConstraintTemplate is in the policy. Plus, the behavior would differ based on whether the ConstraintTemplate were before or after the Constraints, which I'm not sure is desirable. What do you think?

Reply:
Yeah, I think that is sufficient. I agree, it wasn't worth the value of adding that additional code.